File: jplag.frontend.chars/src/main/java/de/jplag/chars/Language.java
package de.jplag.chars;

import java.io.File;

import org.kohsuke.MetaInfServices;

import de.jplag.TokenList;

/*
 * read in text files as characters
 */
@MetaInfServices(de.jplag.Language.class)
public class Language implements de.jplag.Language {

    public static final String IDENTIFIER = "char";
    private final Parser parser;

    public Language() {
        parser = new Parser();
    }

    @Override
    public String[] suffixes() {
        return new String[] {".TXT", ".txt", ".ASC", ".asc", ".TEX", ".tex"};
    }

    @Override
    public String getName() {
        return "Character Parser";
    }

    @Override
    public String getIdentifier() {
        return IDENTIFIER;
    }

    @Override
    public int minimumTokenMatch() {
        return 10;
    }

    @Override
    public TokenList parse(File dir, String[] files) {
        return this.parser.parse(dir, files);
    }

    @Override
    public boolean hasErrors() {
        return this.parser.hasErrors();
    }

    @Override
    public boolean supportsColumns() {
        return false;
    }

    @Override
    public boolean isPreformatted() {
        return false;
    }
}

File: jplag.frontend.scala/src/main/scala/de/jplag/scala/Language.scala
package de.jplag.scala

import de.jplag.TokenList

import java.io.File

import org.kohsuke.MetaInfServices

class Language extends de.jplag.Language {
  private val parser = new Parser
  private final val fileExtensions = Array(".scala", ".sc")

  override def suffixes: Array[String] = fileExtensions

  override def getName = "Scala parser"

  override def getIdentifier = "scala"

  override def minimumTokenMatch = 8

  override def parse(dir: File, files: Array[String]): TokenList = this.parser.parse(dir, files)

  override def hasErrors: Boolean = this.parser.hasErrors

}
